modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ScannableTableImpl.java

```diff
@@ -63,6 +72,113 @@ public <RowT> Publisher<RowT> scan(ExecutionContext<RowT> ctx, PartitionWithTerm
             pub = internalTable.scan(partWithTerm.partId(), txAttributes.id(), recipient, null, null, null, 0, null);
         }
 
-        return StorageScanNode.convertPublisher(pub, (item) -> rowConverter.toRow(ctx, item, rowFactory, requiredColumns));
+        return new TransformingPublisher<>(pub, item -> rowConverter.toRow(ctx, item, rowFactory, requiredColumns));
     }
+
+    /** {@inheritDoc} */
+    @Override
+    public <RowT> Publisher<RowT> indexRangeScan(ExecutionContext<RowT> ctx, PartitionWithTerm partWithTerm, RowFactory<RowT> rowFactory,
+            int indexId, String indexName, List<String> columns, @Nullable RangeCondition<RowT> cond, @Nullable BitSet requiredColumns) {
```

Comment:
```suggestion
    public <RowT> Publisher<RowT> indexRangeScan(
            ExecutionContext<RowT> ctx, 
            PartitionWithTerm partWithTerm,
            RowFactory<RowT> rowFactory,
            int indexId, 
            String indexName, 
            List<String> columns, 
            @Nullable RangeCondition<RowT> cond, 
            @Nullable BitSet requiredColumns
    ) {
```

Reply:
btw, `indexName` is not used

Reply:
@AMashenkov I added indexName to simplify testing  because there is no way to connect `indexId` and a call index scan/lookup methods (yes, there are tests in the main branch that intercept index scans and we are going to need this anyways)). 
An alternative would require moving index scans/lookup to a separate class + making all the necessary changes to make it accessible from `LogicalRelImplementor::visit(IgniteIndexScan rel)`. I think this approach would create a lot of unnecessary complexity.

Reply:
@AMashenkov See  runner / `ItSecondaryIndexTest testNullsInSearchRow`.

Reply:
@AMashenkov Replaced ItSecondaryIndexTest tests that with unit tests for `ExpressionFactoryImplTest` that validate nulls / null bounds.